get-human-redable-permissions-for-RBAC-groups.groovy

```diff
@@ -0,0 +1,31 @@
+import nectar.plugins.rbac.strategy.RoleMatrixAuthorizationStrategyImpl;
+import nectar.plugins.rbac.strategy.RoleMatrixAuthorizationConfig;
+import nectar.plugins.rbac.strategy.RoleMatrixAuthorizationPlugin;
+import hudson.security.PermissionGroup;
+import nectar.plugins.rbac.roles.Role;
+
+//Obtain security configuration
+RoleMatrixAuthorizationStrategyImpl strategy = RoleMatrixAuthorizationStrategyImpl.getInstance()
+RoleMatrixAuthorizationConfig config = RoleMatrixAuthorizationPlugin.getConfig()
+
+def map = [:]
+
+//Populate map
+PermissionGroup.getAll().each { permissionGroup ->
+  permissionGroup.permissions.each { permission ->
+    if (permission.enabled && permissionGroup.id != "N/A") {
+      map[permission.id] = permissionGroup.id + '/' + permission.name
+    }
+  }
+}
+
+config.getRoles().each{r ->
+    DESCRIPTION = r
+    Role rc = new Role(r)
+
+    rc.getPermissionProxies().each{p ->
+        println DESCRIPTION + ' - ' + map[p.id]
+    }
+}
+
+return null
```

Comment:
No need to populate a map. That map will probably miss some permissions when run from OC. because some permissions are inferred by RBAC from Controllers. 

You get all the information you need from the Role object:

```suggestion
import nectar.plugins.rbac.strategy.RoleMatrixAuthorizationConfig;
import nectar.plugins.rbac.strategy.RoleMatrixAuthorizationPlugin;
import nectar.plugins.rbac.roles.Role;

RoleMatrixAuthorizationPlugin.getConfig().getRoles().each{r ->
    new Role(r).getPermissionProxies().each{p ->
        println r + " - " + p.group.title + '/' + p.name
    }
}

return null
```

Reply:
Wow, incredible suggestion! I should have thought about the entity Role exist and how to extract its information (in case of existence). Much easier than mapping them by hand the permissions, and more if you say that some of them are inferred.